docs/scripting-element-object.rst

```diff
@@ -0,0 +1,393 @@
+call; ``visible`` provides an information about error type; otherwise ``visible`` indicated whether
+the element is visible.
+
+**Async:** no.
+
+
+.. _splash-element-fetch-text:
+
+element:fetch_text
+------------------
+
+Fetch a text information from the element
+
+**Signature:** ``ok, visible = element:fetch_text()``
+
+**Returns:** ``ok, text`` pair. If ``ok`` is nil then error happened during the function call;
+``text`` provides an information about error type; otherwise ``text`` is a text content
+of the element.
+
+**Async:** no.
+
+It tries to return the value of the following JavaScript ``Node`` properties:
+
+* textContent
+* innerText
+* value
+
+If all of them are empty an empty string is returned.
+
+
+.. _splash-element-info:
+
+element:info
+------------
+
+Get useful information about the element.
+
+**Signature:** ``ok, info = element:info()``
+
+**Returns:** ``ok, info`` pair. If ``ok`` is nil then error happened during the function call;
+``info`` provides an information about error type; otherwise ``info`` is a table with info.
+
+**Async:** no.
+
+Info is a table with the following fields:
+
+* nodeName - node name in a lower case (e.g. *h1*)
+* attributes - table with attributes names and its values
+* tag - html string representation of the element
+* html - inner html of the element
+* text - inner text of the element
+* x - x coordinate of the element
+* y - y coordinate of the element
+* width - width of the element
+* height - height of the element
+* visible - flag representing if the element is visible
+
+
+.. _splash-field-value:
+
+element:field_value
+-------------------
+
+Get value of the field element (input, select).
+
+**Signature:** ``ok, info = element:field_value()``
+
+**Returns:** ``ok, value`` pair. If ``ok`` is nil then error happened during the function call;
+``value`` provides an information about error type; otherwise ``value`` is a value of the
+element.
+
+**Async:** no.
+
+The value can be a
+* string - for text/radio input and select and other element types
+* array of strings - for multi select
+* boolean - for checkbox input
+
+
+.. _splash-form-values:
+
+element:form_values
+-------------------
+
+Return a table with form values if the element type is *form*
+
+**Signature:** ``ok, info = element:form_values()``
+
+**Returns:** ``ok, values`` pair. If ``ok`` is nil then error happened during the function call
+or node type is not *form*; ``values`` provides an information about error type; otherwise
+``values`` is a table of values.
+
+**Async:** no.
+
+
+.. _splash-send-keys:
+
+element:send_keys
+-----------------
+
+Send keyboard events to the element.
+
+**Signature:** ``ok = element:send_keys(keys)``
+
+**Parameters**
+
+* keys - string representing the keys to be sent as keyboard events.
+
+**Returns:** ``ok`` pair. If ``ok`` is nil then error happened during the function call.
+
+**Async:** no.
+
+This methods do the following:
+
+* checks whether the selected element is editable
```

Comment:
E.g. element may become editable only after a click, or it may listen to keyboard events without being editable